src/contract/allocate_output.jl

```diff
@@ -1,5 +1,18 @@
 using Base.PermutedDimsArrays: genperm
 
+function checkndims(a1, labels1, a2, labels2)
```

Comment:
```suggestion
function checkndims_contract(a1, labels1, a2, labels2)
```
or something like that, to make it clear that the check is specific to contraction.

Going further with that, I like the MatrixAlgebraKit.jl interface for checking inputs based around a generic `check_input` function: https://github.com/QuantumKitHub/MatrixAlgebraKit.jl/blob/v0.2.4/src/implementations/svd.jl#L11. So here that might look like:
```suggestion
function check_input(::typeof(contract), a1, labels1, a2, labels2)
```
That would then generalize to checking the inputs of other functions in the library.

Reply:
Done. Currently I define a `TensorAlgebra.check_input`. Would you rather overload `MatrixAlgebraKit.check_input`?

Reply:
I would lean towards keeping it as `TensorAlgebra.check_input` since the function is a tensor algebra function, not a matrix factorization function.